### PR TITLE
Closes #11: fix(publish): fix publishing request

### LINKED ISSUE #11: 400 Error when publishing
As title suggests I'm getting a 400 error when publishing to gitlab. I think this is due to 
`    body: {tag_name: gitTag, ref: gitHead, release_description: notes}, // eslint-disable-line camelcase
` in `publish.js`

I think this is submitting the incorrect payload. As outlined in the gitlab docs:
- id (required) - The ID or URL-encoded path of the project owned by the authenticated user
- tag_name (required) - The name of a tag
- description (required) - Release notes with markdown support`

Is this the case or am I totally off? Happy to do a PR for this if I am correct - just wanted to know if I was correct in my guess.
Thanks for the plugin! :D

### PR DESCRIPTION
**Pull request by DewarM**
## Closes #11: fix(publish): fix publishing request

Change HTTP verb from PUT to POST. Remove ref, add id and change release_description to description
in body of request.

Update tests in line with changes.

### CODE REVIEW

File: lib/publish.js [hunk 1]
```diff
@@ -14,10 +14,10 @@ module.exports = async (pluginConfig, {repositoryUrl}, {gitHead, gitTag, notes},
   debug('release ref: %o', gitHead);
 
   debug('Update git tag %o with commit %o and release description', gitTag, gitHead);
-  await got.put(urlJoin(apiUrl, `/projects/${repoId}/repository/tags/${gitTag}/release`), {
+  await got.post(urlJoin(apiUrl, `/projects/${repoId}/repository/tags/${gitTag}/release`), {
     json: true,
     headers: {'PRIVATE-TOKEN': gitlabToken},
-    body: {tag_name: gitTag, ref: gitHead, release_description: notes}, // eslint-disable-line camelcase
+    body: {id: repoId, tag_name: gitTag, description: notes}, // eslint-disable-line camelcase
```
Comment (pvdlg) [inline lib/publish.js]: The `id` is not required in the payload, only in the URI (`/projects/${repoId}/repository/tags/${gitTag}/release`).
This line should be `body: {tag_name: gitTag, description: notes}`

File: test/integration.test.js [hunk 2]
```diff
@@ -50,17 +50,18 @@ test.serial('Verify GitLab auth', async t => {
 test.serial('Publish a release', async t => {
   const owner = 'test_user';
   const repo = 'test_repo';
+  const repoId = `${owner}%2F${repo}`;
```
Comment (pvdlg) [inline test/integration.test.js]: No need to create this extra variable as `repoId` should be used only in the URI, not in the payload.

File: test/integration.test.js [hunk 3]
```diff
@@ -50,17 +50,18 @@ test.serial('Verify GitLab auth', async t => {
...
   process.env.GL_TOKEN = 'gitlab_token';
   const nextRelease = {gitHead: '123', gitTag: 'v1.0.0', notes: 'Test release note body'};
   const options = {branch: 'master', repositoryUrl: `https://gitlab.com/${owner}/${repo}.git`};
 
   const gitlab = authenticate()
-    .get(`/projects/${owner}%2F${repo}`)
+    .get(`/projects/${repoId}`)
```
Comment (pvdlg) [inline test/integration.test.js]: No need for this change as the `repoId ` is used only in the URI not in the payload.

**Comment by pvdlg** [comment]
Thanks for the PR. I added inline comment for the few changes necessary.

**Comment by DewarM** [comment]
@pvdlg Awesome, added your requested changes. I misread the API docs.

**Comment by pvdlg** [comment]
Thanks!

**Comment by DewarM** [comment]
@pvdlg Thanks for your help :)

**Pull request merged**